### PR TITLE
drivers: i2s_mcux_sai: Cleanup

### PR DESCRIPTION
**Pull request by decsny**
While debugging the i2s sai driver, we discovered that the driver is very hard to follow and there is a lot of redundant code. Clean up the code to be more readable and understandable. All the logic should be the same.

### CODE REVIEW

**Comment by decsny** [comment]
rebased, updated the PR, removed some of the changes. The changes in the current state of the PR reduce the size of the driver .text by almost 3%.

**Review by danieldegrasse — DISMISSED** [review]
Approving, but hardware testing should be done before this merges

**Review by bjarki-andreasen — DISMISSED** [review]

**Comment by kartben** [comment]
Needs rebase

**Comment by decsny** [comment]
removed the commit refactoring the token names, it was not necessary and too many conflicts, I didn't care

also tested on RT1060 EVKB and passed

File: drivers/i2s/i2s_mcux_sai.c [hunk 1]
```diff
@@ -289,35 +289,30 @@ static void i2s_dma_tx_callback(const struct device *dma_dev, void *arg, uint32_
...
+	/* all DMA blocks are free but no blocks were queued */
+	if (strm->state == I2S_STATE_STOPPING) {
+		/* TX queue has drained */
+		strm->state = I2S_STATE_READY;
+		LOG_DBG("TX stream has stopped");
+	} else {
+		strm->state = I2S_STATE_ERROR;
+		LOG_ERR("TX Failed to reload DMA");
 	}
+	goto disabled_exit_no_drop;
```
Comment (bjarki-andreasen) [inline drivers/i2s/i2s_mcux_sai.c:315]: I don't think this line is required, its essentially saying "jump to next line" which I'm fairly sure it was going to do anyway :)

**Review by bjarki-andreasen — APPROVED** [review]
Really nice! ~found one oddity~ :)

File: drivers/i2s/i2s_mcux_sai.c (hunk 1 above)
Comment (decsny) [reply drivers/i2s/i2s_mcux_sai.c:315]: yes, it's not required, I put it there to be explicit, for future contributor, so that there are no accidental bugs from misunderstanding any logic. maybe it's silly but I thought it was good to be explicit about all the points where we go to any exit in the function

**Review by dleach02 — APPROVED** [review]
